src/Common/src/Interop/Windows/Winsock/Interop.WSARecv.cs

```diff
@@ -40,15 +40,15 @@ internal static partial class Winsock
 
         internal static unsafe SocketError WSARecv(
             IntPtr socketHandle,
-            WSABuffer[] buffers,
+            Span<WSABuffer> buffers,
             int bufferCount,
             out int bytesTransferred,
             ref SocketFlags socketFlags,
             NativeOverlapped* overlapped,
             IntPtr completionRoutine)
         {
-            Debug.Assert(buffers != null && buffers.Length > 0 );
-            fixed (WSABuffer* buffersPtr = &buffers[0])
+            Debug.Assert(!buffers.IsEmpty);
+            fixed (WSABuffer* buffersPtr = &buffers[0]) // deliberately not using GetReference - want this to explode if empty
```

Comment:
Why wouldn't we want to use GetReference?  By construction it can't be empty here, which is why there's an assert.  If it were empty, bufferCount should be 0, in which case it should be fine if we pass a garbage pointer to WSARecv as it'll never be dereferenced.  And indexing into the span incurs an additional cost in the form of a bounds check, whereas GetReference doesn't.  This just seems like unnecessary overhead.

Reply:
@stephentoub  happy to change it if you reckon; my primary aim there was to ensure that *if* it could ever be empty by some code path I missed, then the old code and new code would fail in the same way, rather than by passing an empty pointer